sdk/client/clientv2.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018, 2019 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools"
+	"github.com/sirupsen/logrus"
+	"os"
+	"strings"
+
+	"github.com/networkservicemesh/networkservicemesh/sdk/common"
+)
+
+const (
+	annotationEnv = "NS_NETWORKSERVICEMESH_IO"
+)
+
+type NsmClientV2 struct {
+	clients     []*NsmClient
+	connections []*connection.Connection
+}
+
+func configFromUrl(configuration *common.NSConfiguration, url *tools.NsUrl) *common.NSConfiguration {
+	var conf common.NSConfiguration
+	if configuration != nil {
+		conf = *configuration
+	}
+	conf.OutgoingNscName = url.NsName
+	var labels strings.Builder
+	separator := false
+	for k, v := range url.Params {
+		if separator {
+			labels.WriteRune(',')
+		} else {
+			separator = true
+		}
+		labels.WriteString(k)
+		labels.WriteRune('=')
+		labels.WriteString(v[0])
+	}
+	conf.OutgoingNscLabels = labels.String()
+	return &conf
+}
+
+func NewNSMClientV2(ctx context.Context, configuration *common.NSConfiguration) (*NsmClientV2, error) {
```

Comment:
I don't see this as a second version fo the client. It is a higher abstraction that builds on top of the client. My preference is that this one is renamed to `NewNSMClientList` or similar.